### PR TITLE
Remove digest from cache then we remove blob

### PR DESCRIPTION
**Pull request by legionus**

### CODE REVIEW

File: pkg/dockerregistry/server/admin.go [hunk 1]
```diff
@@ -81,9 +84,14 @@ func (bh *blobHandler) Delete(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	err := bh.Cache.Remove(bh.Digest)
+	if err != nil {
+		dcontext.GetLogger(bh).Errorf("blobHandler: ignore error: unable to remove %q from cache: %v", bh.Digest, err)
```
Comment (bparees) [inline pkg/dockerregistry/server/admin.go]: is this going to log an error in cases where the digest simply didn't exist in the cache?

Comment (bparees) [reply pkg/dockerregistry/server/admin.go]: @legionus please be sure to answer this question ^^

Comment (dmage) [reply pkg/dockerregistry/server/admin.go]: @bparees the current cache implementation returns an error only when the digest is invalid, which should never happen.

Comment (bparees) [reply pkg/dockerregistry/server/admin.go]: thanks

**Pull request merged**